app/screens/gallery/pager/gestures/usePagerPanGesture.ts

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {Gesture, type GestureStateChangeEvent, type GestureUpdateEvent, type PanGestureHandlerEventPayload} from 'react-native-gesture-handler';
+import {useSharedValue, withSpring} from 'react-native-reanimated';
+
+import {MAX_VELOCITY, MIN_VELOCITY} from '@constants/gallery';
+import {pagerPanSpringConfig} from '@screens/gallery/animation_config/spring';
+import {useLightboxSharedValues} from '@screens/gallery/lightbox_swipeout/context';
+import {clampVelocity, friction} from '@utils/gallery';
+
+import {usePagerSharedValues} from '../context';
+
+export default function usePagerPanGesture() {
+    const {
+        isActive,
+        isPagerInProgress,
+        velocity,
+        getCanSwipe,
+        getNextIndex,
+        pagerX,
+        sharedWidth,
+        toValueAnimation,
+        getPageTranslate,
+        index,
+        onIndexChange,
+    } = usePagerSharedValues();
+    const {allowsOtherGestures} = useLightboxSharedValues();
+
+    const offset = useSharedValue<number | null>(null);
+
+    const shouldHandleEvent = (evt: GestureStateChangeEvent<PanGestureHandlerEventPayload> | GestureUpdateEvent<PanGestureHandlerEventPayload>) => {
+        'worklet';
+        return (
+            allowsOtherGestures() &&
+                evt.numberOfPointers === 1 &&
+                isActive.value &&
+                Math.abs(evt.velocityX) > Math.abs(evt.velocityY)
+        ) || isPagerInProgress.value;
```

Comment:
I'm trying to get a handle on how the pager works, so forgive me if this is a naive question. What happens if pager is in progress, but multiple fingers are touching, and/or vertical swipe? What gets priority? Do they all activate?

Reply:
Cool, glad it helped